sui/deploy-contract.js

```diff
@@ -23,49 +54,179 @@ async function processCommand(contractName, config, chain, options) {
+
+    if (!chain.contracts[contractName]) {
+        throw new Error(`Cannot find specified contract: ${contractName}`);
+    }
+
+    const contractsConfig = chain.contracts;
+    const packageConfig = contractsConfig?.[contractName];
+
+    validateParameters({ isNonEmptyString: { contractName } });
+
+    if (packageDependencies) {
+        for (const dependencies of packageDependencies) {
+            const packageId = contractsConfig[dependencies]?.address;
+            updateMoveToml(dependencies, packageId);
+        }
+    }
+
+    const builder = new TxBuilder(client);
+    await upgradePackage(client, keypair, contractName, packageConfig, builder, options);
+}
+
+async function mainProcessor(args, options, processor) {
     const config = loadSuiConfig(options.env);
-    await processor(contractName, config, config.sui, options);
+    await processor(...args, config, config.sui, options);
     saveConfig(config, options.env);
+
+    if (options.offline) {
+        const { txFilePath } = options;
+        validateParameters({ isNonEmptyString: { txFilePath } });
+
+        const txB64Bytes = toB64(options.txBytes);
+
+        writeJSON({ message: options.offlineMessage, status: 'PENDING', unsignedTx: txB64Bytes }, txFilePath);
+        printInfo(`Unsigned transaction`, txFilePath);
+    }
 }
 
 if (require.main === module) {
     const program = new Command();
 
-    program
-        .name('deploy-contract')
-        .addOption(new Option('--packageName <packageName>', 'Package name to deploy'))
-        .addArgument(new Argument('<contractName>', 'Contract name to deploy').choices(Object.keys(contractMap)))
-        .description('Deploy SUI modules');
-
-    addBaseOptions(program);
-
-    program.action((contractName, options) => {
-        mainProcessor(contractName, options, processCommand);
-    });
+    program.name('deploy-contract').description('Deploy/Upgrade packages');
+
+    const deployCMD = program
+        .name('deploy')
+        .description('Deploy SUI modules')
+        .command('deploy <contractName>')
+        .addOption(new Option('--signers <signers>', 'JSON with the initial signer set').env('SIGNERS'))
+        .addOption(new Option('--operator <operator>', 'operator for the gateway (defaults to the deployer address)').env('OPERATOR'))
+        .addOption(new Option('--minimumRotationDelay <minimumRotationDelay>', 'minium delay for signer rotations (in ms)').default('0'))
+        .addOption(new Option('--domainSeparator <domainSeparator>', 'domain separator').default(HashZero))
+        .addOption(new Option('--nonce <nonce>', 'nonce for the signer (defaults to HashZero)'))
+        .addOption(new Option('--previousSigners <previousSigners>', 'number of previous signers to retain').default('15'))
+        .addOption(
+            new Option('--policy <policy>', 'upgrade policy for upgrade cap: For example, use "0" to allow all types of upgrades')
+                .choices(['0', '128', '192'])
```

Comment:
use strings for readable policy names. Convert to an int internally (maybe sui sdk has a helper for conversion?)